IHE DEC Gateway - first draft ready for review (#163)
IHE/sdpi-fhir: SDPi_Supplement/asciidoc/volume2/gateways/tf2-ch-b-gateway-acm.adoc

=== SDPi ACM Gateway -- Mapping

NOTE:  This file contains the SDPi gateway actor (SOMDS_ACM_Gateway) logic for mapping from SDC BICEPS MDIB content and interactions with IHE DEV "PCD" ACM profile messaging.

==== Scope

> This chapter defines the mapping from SDC MDIB content as defined in this document and its underlying standards, to IHE Alert Communication Management (ACM) profile messages as defined in the <<ihe_pcd_tf-2_2019>>.

Author: I have no strong opinion - just define some guidelines.

The SDPi gateway actor represents the Alarm Reporter (AR) role of the IHE ACM profile.

> The following sections supplement the IHE ACM profile as appropriate. If there are no supplementing definitions, the definitions as described in the <<ihe_pcd_tf-2_2019>> will apply.

Reviewer: ```suggestion
The following sections supplement the IHE ACM profile as appropriate. If there are no supplementing definitions, the definitions as described in the <<ihe_pcd_tf_2_2019>> will apply.
```
Please fully underscore.

Author: done - changed in all files.

==== Referenced Standards & Profiles
This section provides an overview about the referenced standards and profiles used in this chapter:

* <<ihe_pcd_tf_2_2019>>
* <<ieee_11073_10702_202x>>
* <<ieee_11073_10701_2022>>
* <<ieee_11073_10700_2022>>
* <<ieee_11073_10207_2017>>

==== HL7 Segment Descriptions
The following sections refer to the *Appendix B Common Segment Descriptions* of the <<ihe_pcd_tf_2_2019>>.

> include::tf2-ch-b-gateway-msh-mapping.adoc[]

Reviewer: This file is included twice in the document (the other one in SDPi_Supplement/asciidoc/volume2/gateways/tf2-ch-b-gateway-dec.adoc). This leads to a requirement duplicate. Which include is incorrect?

Author: Sh...! This requirement is needed for the ACM and DEC. The same is true for the PID and PV1 and OBX-18.
Any suggestions?

Author: Could you check if a requirement is only defined in one file (do not check the final file with all includes)?

> include::tf2-ch-b-gateway-pid-mapping.adoc[]

Author: Sh...! This requirement is needed for the ACM and DEC. The same is true for the PID and PV1 and OBX-18.
Any suggestions?

> include::tf2-ch-b-gateway-pid-mapping.adoc[]

Author: Could you check if a requirement is only defined in one file (do not check the final file with all includes)?

include::tf2-ch-b-gateway-pv1-mapping.adoc[]

===== OBR - Observation Request Segment

> The HL7 Observation Request (OBR) segment requires a mapping from the SDC containment tree and metric data to the OBR segment fields.

Reviewer: ```suggestion
The HL7 Observation Request (OBR) segment requires a mapping from the <<acronym_mdib>> containment tree and metric data to the OBR segment fields.
```

Author: Please ignore - ACM is not part of the review.

> The the HL7 Obervation Request (OBR) segment requires a mapping from the SDC containment tree and metric data to the OBR segment fields.

Author: Fixed

> include::tf2-ch-b-gateway-pv1-mapping.adoc[]

Reviewer: Included twice as well. See https://github.com/IHE/sdpi-fhir/pull/163/files#r1039348454

Author: Sh...! This requirement is needed for the ACM and DEC. The same is true for the PID and PV1 and OBX-18.
Any suggestions?

Author: Could you check if a requirement is only defined in one file (do not check the final file with all includes)?

====== OBR-2 Placer Order Number

> For IHE ACM profile, the OBR-2 field shall contain the identifier of the Alarm Reporter (AR) of the IHE ACM gateway (not the individual device identifier). For further information, please refer to the <<ihe_pcd_tf-2_2019>>.

Author: Fixed.

> For the IHE ACM profile, the OBR-2 field shall contain the identifier of the Alarm Reporter (AR) of the IHE ACM gateway (not the individual device identifier). For further information, please refer to the <<ihe_pcd_tf-2_2019>>.

Reviewer: Is this supposed to be an sdpi_requirement box as it contains a "shall" keword?

```suggestion
.RXXXX
[sdpi_requirement#rxxxx,sdpi_req_level=shall]
****
For the IHE ACM profile, the OBR-2 field shall contain the identifier of the Alarm Reporter (AR) of the IHE ACM gateway (not the individual device identifier).

.Notes
[%collapsible]
====
NOTE: For further information, please refer to the <<ihe_pcd_tf-2_2019>>.
====
****
```

Author: done.

====== OBR-3 Filler Order Number

> For the IHE ACM profile, the OBR-3 field shall contain the identifier of the Alarm Reporter (AR) of the IHE ACM gateway (not the individual device identifier). For further information, please refer to the <<ihe_pcd_tf-2_2019>>.

Reviewer: See https://github.com/IHE/sdpi-fhir/pull/163/files#r1012884811

Author: done.
